docs/source/llms/langchain/index.rst

```diff
@@ -550,3 +550,44 @@ string response, which can be compared to the `"ground_truth"` column.
 
 For a complete example of a LangGraph model that works with this evaluation example, see the 
 `MLflow LangGraph blog <https://mlflow.org/blog/langgraph-model-from-code>`_.
+
+How to control whether my input is converted to List[langchain.schema.BaseMessage] in PyFunc predict?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If user provides inputs like ``{"messages": [{"role": "user", "content": "some_question"}]}``, MLflow internally tries to
+convert the input to List[langchain.schema.BaseMessage] like ``[HumanMessage(content="some_question")]``. If the model is
+AgentExecutor or the model's input_schema contains "messages" field, then MLflow doesn't convert the inputs.
+There're some edge cases where user wants to provide the input as is, in such cases user can set the environment variable
+``MLFLOW_CONVERT_MESSAGES_DICT_TO_FOR_LANGCHAIN`` to ``False``. This will prevent MLflow from 
+implementing such conversion.
+
+.. code-block:: python
+
+    import json
+    import mlflow
+    import os
+    from operator import itemgetter
+    from langchain.schema.runnable import RunnablePassthrough
+
+    model = RunnablePassthrough.assign(
+        problem=lambda x: json.loads(x["messages"][-1]["content"])["problem"]
```

Comment:
Can we simplify this?

```suggestion
        problem=lambda x: {"problem": x}
```

Reply:
Nope, this is the example where the model expects a `{"messages": [{"role": ..., "content": ...}]}` input but shouldn't be converted to List[BaseMessage]